borg/archive.py

```diff
@@ -730,6 +742,24 @@ def identify_key(self, repository):
         cdata = repository.get(next(self.chunks.iteritems())[0])
         return key_factory(repository, cdata)
 
+    def integrity_verification(self):
+        logger.info('Starting integrity verification...')
+        pi = ProgressIndicatorPercent(total=len(self.chunks), msg="Verifying data %6.2f%%", step=0.01, same_line=True)
+        count = 0
+        for chunk_id, (refcount, *_) in self.chunks.iteritems():
+            pi.show()
+            if not refcount:
+                continue
+            encrypted_data = self.repository.get(chunk_id)
+            try:
+                _, data = self.key.decrypt(chunk_id, encrypted_data)
+            except IntegrityError as integrity_error:
+                self.error_found = True
+                logger.error('chunk %s, integrity error: %s', hexlify(chunk_id), integrity_error)
+            count += 1
+        pi.finish()
+        logger.info('Finished integrity verification, verified %d chunks.', count)
```

Comment:
report() is internal to rebuild_refcounts; it uses a format that explicitly references the chunk position inside an item which does not apply here.

I changed the final information message to include total and errored chunk counts which should avoid any ambiguity.